packages/status-js/src/client/member.ts

```diff
@@ -0,0 +1,17 @@
+import { generateUsername } from '../utils/generate-username'
+import { publicKeyToColorHash } from '../utils/public-key-to-color-hash'
+
+import type { ColorHash } from '../utils/public-key-to-color-hash'
+
+export class Member {
+  publicKey: string
+  username: string
+  colorHash: ColorHash
+
+  constructor(publicKey: string) {
+    this.publicKey = publicKey
+    this.username = generateUsername(publicKey)
+    // TODO: can it fail?
+    this.colorHash = publicKeyToColorHash(publicKey)!
```

Comment:
Non-blocking: only if the network broadcasted members with invalid keys, I think. In which case, `status-js` would handle it by returning `undefined`. UI, on the other hand could display a warning icon next to avatar instead.